### ISSUE TITLE
cannot resolve 'file' or 'directory' webpack

### ISSUE THREAD

**Issue by sunny-rajkotiya** [vuejs/vue-cli#19]
I just npm installed vue-cli and did same task as instructed  npm run dev is working fine but when i do  npm run build i got following error message 
```
> vue-project@ build /var/www/html/vue-lab2/vue-project
> rimraf dist && webpack --progress --hide-modules --config build/webpack.prod.conf.js

Hash: f8302fe592cda5dae1c2  
Version: webpack 1.12.11
Time: 2888ms
                          Asset       Size  Chunks             Chunk Names
    app.1c5e9c5cf5fc998a31b4.js    89.4 kB       0  [emitted]  app
app.1c5e9c5cf5fc998a31b4.js.map     687 kB       0  [emitted]  app
                  ../index.html  207 bytes          [emitted]  

ERROR in ./src/App.vue
Module not found: Error: Cannot resolve 'file' or 'directory' ./../../../../../../vue-lab2/vue-project/node_modules/extract-text-webpack-plugin/loader.js in /var/www/html/vue-lab2/vue-project/src
 @ ./src/App.vue 2:0-352
```

this is my package.json

```
{
  "name": "vue-project",
  "description": "This is to test basic vue structure",
  "author": "Sunny Rajkotiya <sunny.rajkotiya@knolskape.com>",
  "private": true,
  "scripts": {
    "dev": "webpack-dev-server --inline --hot --config build/webpack.dev.conf.js",
    "build": "rimraf dist && webpack --progress --hide-modules --config build/webpack.prod.conf.js",
    "test": "karma start build/karma.conf.js --single-run"
  },
  "dependencies": {
    "vue": "^1.0.0"
  },
  "devDependencies": {
    "babel-core": "^6.0.0",
    "babel-loader": "^6.0.0",
    "babel-plugin-transform-runtime": "^6.0.0",
    "babel-preset-es2015": "^6.0.0",
    "babel-preset-stage-2": "^6.0.0",
    "babel-runtime": "^5.8.0",
    "css-loader": "^0.23.0",
    "eslint": "^1.10.3",
    "eslint-friendly-formatter": "^1.2.2",
    "eslint-loader": "^1.2.0",
    "extract-text-webpack-plugin": "^0.9.1",
    "file-loader": "^0.8.4",
    "function-bind": "^1.0.2",
    "html-webpack-plugin": "^1.7.0",
    "inject-loader": "^2.0.1",
    "jasmine-core": "^2.4.1",
    "json-loader": "^0.5.4",
    "karma": "^0.13.15",
    "karma-jasmine": "^0.3.6",
    "karma-phantomjs-launcher": "^0.2.1",
    "karma-spec-reporter": "0.0.23",
    "karma-webpack": "^1.7.0",
    "phantomjs": "^1.9.19",
    "rimraf": "^2.5.0",
    "url-loader": "^0.5.7",
    "vue-hot-reload-api": "^1.2.0",
    "vue-html-loader": "^1.0.0",
    "vue-loader": "^8.0.0",
    "vue-style-loader": "^1.0.0",
    "webpack": "^1.12.2",
    "webpack-dev-server": "^1.12.0"
  }
}

```
i have npm version 3.3.12

I  tried all solutions mentioned in https://github.com/vuejs/vue-loader/issues/70 ,https://github.com/webpack/webpack/issues/981, also tried adding resolve.root = resolve.path(__dirname,"node_modules") or resolveLoader: {                                                                                
    root: path.join(__dirname, 'node_modules')                                                  
} 
but nothing worked

**Comment by voigt** [comment]
I have the same issue. Below you can find the `webpack` output with `--display-error-details`.

```
$node_modules/rimraf/bin.js dist && node_modules/webpack/bin/webpack.js --display-error-details --progress --hide-modules --config build/webpack.prod.Hash: f72ff4065014fb6eddbc                  
Version: webpack 1.12.13
Time: 10075ms
                          Asset       Size  Chunks             Chunk Names
    app.69de1713ea9bc16860f7.js     103 kB       0  [emitted]  app
app.69de1713ea9bc16860f7.js.map     855 kB       0  [emitted]  app
                  ../index.html  183 bytes          [emitted]  

ERROR in ./src/components/ClanView.vue
Module not found: Error: Cannot resolve 'file' or 'directory' ./../../../../../../../../coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js in /var/www/virtual/cv/html/coc_clan_visualizer_ui/src/components
resolve file
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.webpack-loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.web-loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.js doesn't exist
resolve directory
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js doesn't exist (directory default file)
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js/package.json doesn't exist (directory description file)
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.webpack-loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.web-loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.js]
 @ ./src/components/ClanView.vue 2:0-343

ERROR in ./src/components/Member.vue
Module not found: Error: Cannot resolve 'file' or 'directory' ./../../../../../../../../coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js in /var/www/virtual/cv/html/coc_clan_visualizer_ui/src/components
resolve file
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.webpack-loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.web-loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.js doesn't exist
resolve directory
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js doesn't exist (directory default file)
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js/package.json doesn't exist (directory description file)
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.webpack-loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.web-loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.js]
 @ ./src/components/Member.vue 2:0-341

ERROR in ./src/App.vue
Module not found: Error: Cannot resolve 'file' or 'directory' ./../../../../../../../coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js in /var/www/virtual/cv/html/coc_clan_visualizer_ui/src
resolve file
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.webpack-loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.web-loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.loader.js doesn't exist
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.js doesn't exist
resolve directory
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js doesn't exist (directory default file)
  /coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js/package.json doesn't exist (directory description file)
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.webpack-loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.web-loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.loader.js]
[/coc_clan_visualizer_ui/node_modules/extract-text-webpack-plugin/loader.js.js]
 @ ./src/App.vue 2:0-326

ERROR in ./src/components/MemberView.vue
Module not found: Error: Cannot resolve 'file' or 'directory' ./../../.. in /var/www/virtual/cv/html/coc_clan_visualizer_ui/src/components
resolve file
  /var/www/virtual/cv/html is not a file
  /var/www/virtual/cv/html.webpack-loader.js doesn't exist
  /var/www/virtual/cv/html.web-loader.js doesn't exist
  /var/www/virtual/cv/html.loader.js doesn't exist
  /var/www/virtual/cv/html.js doesn't exist
resolve directory
  /var/www/virtual/cv/html/package.json doesn't exist (directory description file)
  directory default file index
    resolve file index in /var/www/virtual/cv/html
      /var/www/virtual/cv/html/index doesn't exist
      /var/www/virtual/cv/html/index.webpack-loader.js doesn't exist
      /var/www/virtual/cv/html/index.web-loader.js doesn't exist
      /var/www/virtual/cv/html/index.loader.js doesn't exist
      /var/www/virtual/cv/html/index.js doesn't exist
[/var/www/virtual/cv/html.webpack-loader.js]
[/var/www/virtual/cv/html.web-loader.js]
[/var/www/virtual/cv/html.loader.js]
[/var/www/virtual/cv/html.js]
[/var/www/virtual/cv/html/index]
[/var/www/virtual/cv/html/index.webpack-loader.js]
[/var/www/virtual/cv/html/index.web-loader.js]
[/var/www/virtual/cv/html/index.loader.js]
[/var/www/virtual/cv/html/index.js]
 @ ./src/components/MemberView.vue 2:0-345
Child html-webpack-plugin for "../index.html":
```

To me this looks like webpack/webpack-loader is looking in the wrong directories, as I'm calling `npm run build` from `/home/cv/html/coc_clan_visualizer_ui`.

Here my `package.json` (I added `vue-async-data` and `vue-router`).
```
{
  "name": "coc_ui",
  "description": "bla",
  "author": "Christoph Voigt",
  "scripts": {
    "dev": "node build/dev-server.js",
    "build": "rimraf dist && webpack --progress --hide-modules --config build/webpack.prod.conf.js",
    "test": "karma start build/karma.conf.js --single-run"
  },
  "dependencies": {
    "vue": "^1.0.16"
  },
  "devDependencies": {
    "babel-core": "^6.0.0",
    "babel-loader": "^6.0.0",
    "babel-plugin-transform-runtime": "^6.0.0",
    "babel-preset-es2015": "^6.0.0",
    "babel-preset-stage-2": "^6.0.0",
    "babel-runtime": "^5.8.0",
    "connect-history-api-fallback": "^1.1.0",
    "css-loader": "^0.23.0",
    "eslint": "^1.10.3",
    "eslint-friendly-formatter": "^1.2.2",
    "eslint-loader": "^1.2.0",
    "eventsource-polyfill": "^0.9.6",
    "express": "^4.13.3",
    "extract-text-webpack-plugin": "^1.0.1",
    "file-loader": "^0.8.4",
    "function-bind": "^1.0.2",
    "html-webpack-plugin": "^2.8.1",
    "inject-loader": "^2.0.1",
    "jasmine-core": "^2.4.1",
    "json-loader": "^0.5.4",
    "karma": "^0.13.15",
    "karma-jasmine": "^0.3.6",
    "karma-phantomjs-launcher": "^1.0.0",
    "karma-spec-reporter": "0.0.23",
    "karma-webpack": "^1.7.0",
    "phantomjs-prebuilt": "^2.1.3",
    "rimraf": "^2.5.0",
    "url-loader": "^0.5.7",
    "vue-async-data":"1.0.2",
    "vue-router":"0.7.10",
    "vue-hot-reload-api": "^1.2.0",
    "vue-html-loader": "^1.0.0",
    "vue-loader": "^8.0.0",
    "vue-style-loader": "^1.0.0",
    "webpack": "^1.12.2",
    "webpack-dev-middleware": "^1.4.0",
    "webpack-hot-middleware": "^2.6.0"
  }
}
```

System Details:
```
$ cat /etc/*-release
CentOS release 6.7 (Final)
CentOS release 6.7 (Final)
CentOS release 6.7 (Final)
$ node -v
v5.1.1
$ npm -v
3.3.12
```

Help, hints and ideas very appreciated :)

**Comment by voigt** [comment]
No issues to compile the same on OSX:

```
$ node -v
v5.5.0
$ npm -v
3.5.3
```

**Comment by yyx990803** [comment]
Looks like a webpack bug on Linux?

**Comment by 2eron** [comment]
I have the same problem when pack on Linux:
`
ERROR in ./src/components/Slider.vue
Module not found: Error: Cannot resolve 'file' or 'directory' ./../../../../../html in /root/workspace/moblie/src/components
 @ ./src/components/Slider.vue 2:0-283
`
It's ok on my OSX

**Comment by cnicodeme** [comment]
I confirm I have the same problem under Linux (Fedora 22) when using `npm run build` (but works correctly with `npm run dev`).

Here's my config : 

```bash
$ node -v
v5.5.0
$ npm -v
3.3.12
```

**Comment by yyx990803** [comment]
Actually, since this is a problem specific to the webpack template, I'm moving it over there instead: https://github.com/vuejs-templates/webpack/issues/41

**Comment by cnicodeme** [comment]
Hi tested something on my installation : 

I initiated a new project using vue-cli and webpack, and did `npm install && npm run build` and I got this error : 

> ERROR in ./src/App.vue
Module not found: Error: Cannot resolve 'file' or 'directory' ./../../../../../../test/node_modules/extract-text-webpack-plugin/loader.js in /var/www/html/test/src
 @ ./src/App.vue 2:0-321
Child html-webpack-plugin for "../index.html":

I had this issue without modifying anything.

Maybe that would help, but despite the fact that I've installed vue-cli as root using npm install -g vue-cli, I don't have any `vue` command in my $PATH so I did the following : 

> /opt/node/5.3.0/lib/node_modules/vue-cli/bin/vue init webpack test

And odly, it correspond to the number of .. in the above error message.
Hoping that this could help! :)
